src/core/ipns/validator.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+// Create IPNS Entry data for being signed
+const getDataForSignature = (value, validityType, validity) => {
+  const valueBuffer = Buffer.from(value)
+  const validityTypeBuffer = Buffer.from(validityType.toString())
+  const validityBuffer = Buffer.from(validity)
+
+  return Buffer.concat([valueBuffer, validityTypeBuffer, validityBuffer])
+}
+
+// Sign IPNS Entry for publish
+const sign = (privateKey, value, validityType, validity, callback) => {
+  const dataForSignature = getDataForSignature(value, validityType, validity)
+
+  privateKey.sign(dataForSignature, (err, signature) => {
+    if (err) {
+      return callback(err)
+    }
+    return callback(null, signature)
+  })
+}
+
+// Verify IPNS entry on resolve
+const verify = (publicKey, entry, callback) => {
+  const { value, validityType, validity } = entry
+  const dataForSignature = getDataForSignature(value, validityType, validity)
+
+  // Validate Signature
+  publicKey.verify(dataForSignature, entry.signature, (err, result) => {
+    if (err) {
+      return callback(err)
+    }
+
+    // Validate EOL
+    const validityDate = Date.parse(validity.toString())
```

Comment:
consider using https://date-fns.org/

Reply:
FYI, the only valid time format is RFC3339 with nanosecond precision. That is:

2006-01-02T15:04:05.999999999Z07:00

Reply:
Note: It's extremely important that you only except strings of this exact form. Not validating this has unfortunate security implications.

This also needs to check the validity type.

Reply:
I will support milliseconds precision at first here and I added a task to improve the precision to nanosecond later (in the list above). 

Basically, JS does not support nanoseconds precision natively and all the libs that I managed to find don't support summing date times with nanosecond precision.

Reply:
When you say that I must validate if the string has this exact form, you say that it should be compliant with RFC3339, right?